### PR TITLE
add: script llama-cpp-integration

### PR DESCRIPTION
**Pull request by eljamm**
See https://github.com/qownnotes/scripts/pull/237#issuecomment-2169560502

### CODE REVIEW

**Review by pbek — COMMENTED** [review]
Can you please set yourself as maintainer 😉

**Comment by pbek** [comment]
And use 0.0.1 as version number

**Comment by eljamm** [comment]
Thanks, but I don't want to maintain the script because:

1. I don't know how to write qml
2. I don't think I'll be using it much. I tested it mainly out of curiosity.
3. I just wanted to help you out a bit when you asked 😊

I'll surely contribute again in the future, though, if I think something can be improved or fixed.

File: llama-cpp-integration/llama-cpp-integration.qml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            "description": "The chat base URL of the llama-cpp API.",
+            "type": "string",
+            "default": "http://127.0.0.1:8080/v1/chat/completions",
+        },
+        {
+            "identifier": "models",
+            "name": "Models",
+            "description": "Comma separated list of models to use.",
+            "type": "string",
+            "default": "llama3,gemma:2b",
```
Comment (pbek) [inline llama-cpp-integration/llama-cpp-integration.qml:28]: Which models does llama-cpp support? Is the current example useful?

**Comment by pbek** [comment]
> I don't know how to write qml

It's just JS anyway...

> I don't think I'll be using it much. I tested it mainly out of curiosity.

I implemented the whole OpenAI API integration out of curiosity 😆

File: llama-cpp-integration/llama-cpp-integration.qml (hunk 1 above)
Comment (eljamm) [reply llama-cpp-integration/llama-cpp-integration.qml:28]: The list of supported models can be found [here](https://github.com/ggerganov/llama.cpp#description).

llama3 :llama: and gemma are indeed supported

**Comment by eljamm** [comment]
> I implemented the whole OpenAI API integration out of curiosity 😆

And I really commend you for that 😁

**Comment by pbek** [comment]
Thank you very much!

**Pull request merged**